project_setup.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2025 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import mlrun
+
+
+def setup(project: mlrun.projects.MlrunProject) -> mlrun.projects.MlrunProject:
+    source = project.get_param("source")
+
+    # Set project git/archive source and enable pulling latest code at runtime
+    if source:
+        print(f"Project Source: {source}")
+        project.set_source(project.get_param("source"), pull_at_runtime=True)
+
+    project.set_function(
+        name="prep-data",
+        func="prep_data.py",
+        kind="job",
+        handler="prep_data",
+    )
```

Comment:
I'd add the `gen_iris` function too while you're add it, if we anyway plan to drop project.yaml in the future